### ISSUE TITLE
Panics when a struct contains a enum type

### ISSUE THREAD

**Issue by Y-Nak** [ethereum/fe#794]
### What is wrong?
The code below causes panic 
```
enum Accumulator {
    Zero
    Int(u32)
    Clo(i32, i32)
}

struct MyStruct {
    pub acc: Accumulator
    pub y: i32
}

contract Bar {
}
```
Ref: https://github.com/ethereum/fe/pull/772#discussion_r970844998

@cburgdorf
Suggestion: 
1. I think we still need `#event` attributes in addition to `#index` attributes so a struct can contain enum types.
2. It'd be better to forbid implementing `Emittable` to private structs.

**Comment by Y-Nak** [comment]
>  What would be the benefit of the #event attribute?
It's necessary to allow users to implement a custom `Emittable`/`Encode` trait.

e.g.,
```
struct Foo {
     x: i32
    y: u64
}

impl Emittable for Foo {
    ...
}
```

If `Emittable` is automatically implemented for `Foo`, there is no way to add their custom implementation due to conflicting implementations, right?

**Comment by Y-Nak** [comment]
Oh, I know that it's impossible to implement `Emittable` by the user currently. But will it be allowed in the future, right? Although, the trait name may be changed to `Encode`.

**Comment by cburgdorf** [comment]
>If `Emittable` is automatically implemented for `Foo`, there is no way to add their custom implementation due to conflicting implementations, right?

Ah, right. Except if we decide that a user provided `impl Emittable for X` always precedes the automatically provided one. But yeah, I'm not opposed to the `#event` attribute.

**Comment by Y-Nak** [comment]
> if we decide that a user provided impl Emittable for X always precedes the automatically provided one.

😬 

But for now, I feel inclined to go with your suggestion to keep the change small when assuming we will rewrite the current `Emittable` implementation sooner or later.
> How about we just exclude structs with enum fields from being treated as Emittable for now?

NOTE: It's necessary to exclude structs with non-emittable fields, not only structs with `enum` fields.

e.g.
```rust
enum MyEnum {
}

struct S1 {
    e: MyEnum
}

struct S2 {
    s1: S1
}
```

S2 should be excluded in addition to `S1`.

**Comment by Y-Nak** [comment]
I added `is_encodable` in #796,  it may be available for the fix.

**Comment by Y-Nak** [comment]
resolved in #798
